Enhanced cart

edavis25 opened the pull request:
I created a new cart icon without a black background; however, I removed it from the repo because it looks like it was corrupted like the logo... there must be some issue w/ PNG files. I added the new cart image to the Google Drive under the "Logos & Colors Schemes" folder. When looking at the actual change on the site, you'll have to download that image manually or you'll get the image's alt text (but either way you should be able to see the added badge w/ item count).

edavis25 committed:
Enhanced shopping cart icon

Added a new shopping cart icon without a black background and a "badge" class to framework.css. Modified addtocart/removefromcart functions in the ProductsController to store the total number of items in the cart into session and added the count as a badge shown next to the cart icon when there are 1+ products in cart.

edavis25 committed:
Removed image file

Removed the "cart" image file from the repo as it looked to be corrupted like the logo.

edavis25 commented:
So I thought about this a little more, and we could probably add this functionality with PHP array functions and be able to remove all that logic from the controller. The feature itself obviously is not make or break, but I noticed most eCommerce sites have something similar and it would be a pretty quick enhancement. I'm thinking just using this as an echo statement in the view might work the same:

```
@if (Session::has('shopping_list'))
      <span class="badge">{{ array_sum(array_column(Session::get('shopping_list'), 'qty')) }}</span>
@endif
```

I haven't really played around with this suggestion much to see if its stable, just an idea...let me know what you think and we can either go with what is in this pull request or I can test that snippet for a minute and submit a new request!

mcglonelevi reviewed (changes requested):
You need to change this a little bit.  I will leave some comments.

mcglonelevi commented on laravel/app/Http/Controllers/ProductsController.php:
```diff
@@ -152,7 +152,14 @@ public function addtocart(Request $request, Product $product)
 
+        // Count items in cart
+        $totalItems = 0;
+        foreach ($shoppingList as $p) {
```
Instead of using a foreach, try using a map instead.

http://php.net/manual/en/function.array-map.php

If you don't like using php arrays, you could also use a collection and the collect helper:

https://laravel.com/docs/5.5/collections

mcglonelevi replied:
Actually a reduce would be better.

edavis25 committed:
Refactored enhanced cart logic

Removed the logic counting # of cart items from the products controll/session and refactored it to use an array column sum in the view file.
